scalecube-configuration/src/main/java/io/scalecube/configuration/ConfigurationServiceImpl.java

```diff
@@ -1,133 +1,100 @@
 import io.scalecube.configuration.api.Acknowledgment;
 import io.scalecube.configuration.api.ConfigurationService;
 import io.scalecube.configuration.api.CreateRepositoryRequest;
 import io.scalecube.configuration.api.DeleteRequest;
 import io.scalecube.configuration.api.FetchRequest;
 import io.scalecube.configuration.api.FetchResponse;
 import io.scalecube.configuration.api.SaveRequest;
-import io.scalecube.configuration.authorization.AuthorizationService;
-import io.scalecube.configuration.authorization.OperationType;
-import io.scalecube.configuration.operation.ServiceOperationContext;
-import io.scalecube.configuration.operation.ServiceOperationFactory;
-import io.scalecube.configuration.repository.ConfigurationDataAccess;
-import io.scalecube.configuration.tokens.TokenVerifier;
-import java.util.Objects;
+import io.scalecube.configuration.repository.ConfigurationRepository;
+import io.scalecube.configuration.repository.Document;
+import io.scalecube.configuration.repository.Repository;
+import io.scalecube.security.api.AccessControl;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
-import reactor.core.scheduler.Schedulers;
 
 public class ConfigurationServiceImpl implements ConfigurationService {
 
   private static final Logger logger = LoggerFactory.getLogger(ConfigurationServiceImpl.class);
+  private static final Acknowledgment ACK = new Acknowledgment();
 
-  private final ConfigurationDataAccess dataAccess;
-  private TokenVerifier tokenVerifier;
-  private final AuthorizationService authorizationService;
+  private final ConfigurationRepository repository;
+  private final AccessControl accessControl;
 
-  private ConfigurationServiceImpl(
-      ConfigurationDataAccess dataAccess, TokenVerifier tokenVerifier) {
-    this.dataAccess = dataAccess;
-    this.tokenVerifier = tokenVerifier;
-    this.authorizationService = AuthorizationService.builder().build();
+  public ConfigurationServiceImpl(ConfigurationRepository repository, AccessControl accessContorl) {
+    this.repository = repository;
+    this.accessControl = accessContorl;
   }
 
   @Override
   public Mono<Acknowledgment> createRepository(CreateRepositoryRequest request) {
-    return Mono.fromRunnable(() -> logger.debug("createRepository: enter: request: {}", request))
-        .then(Mono.fromCallable(ServiceOperationFactory::createRepository))
-        .flatMap(operation -> Mono
-            .from(operation.execute(request, context(OperationType.CreateRepoitory))))
-        .subscribeOn(Schedulers.parallel())
-        .doOnSuccess(result -> logger.debug("createRepository: exit: request: {}", request))
-        .doOnError(th -> logger.error("createRepository: request: {}, error: {}", request, th));
+    verifyRequest(request);
+    return accessControl
+        .check(request.token().toString(), ConfigurationService.CONFIG_CREATE_REPO)
```

Comment:
you could pass the whole request  to  `check` so it can check not only the token

Reply:
this will force to implement the interface i am not sure its a good idea